### ISSUE TITLE
Codeception shows SOAP fatal error anyway

### ISSUE THREAD

**Issue by nnrudakov** [Codeception/Codeception#4707]
I'm trying to catch SOAP error when wsdl is wrong.

```php
class Client {
    public function connect() {
        try {
            $client = new \SoapClient('wrong_wsdl');
        } catch (\SoapFault $e) {
            throw new MyException('error');
        }
    }
}
```

I get test OK but SOAP error output too.

```bash
vendor/bin/codecept run unit tests/unit/ClientTest.php:testConnectFail
Codeception PHP Testing Framework v2.3.7
Powered by PHPUnit 6.4.4 by Sebastian Bergmann and contributors.

Unit Tests (1) ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
- WpClientTest: Connect failPHP Fatal error:  SOAP-ERROR: Parsing WSDL: Couldn't load from 'wrong_wsdl' : failed to load external entity "wrong_wsdl"
 in Client.php on line 77
✔ WpClientTest: Connect fail (0.01s)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


Time: 75 ms, Memory: 16.00MB

OK (1 test, 1 assertion)



FATAL ERROR. TESTS NOT FINISHED.
SOAP-ERROR: Parsing WSDL: Couldn't load from 'wrong_wsdl' : failed to load external entity "wrong_wsdl"
 
in Client.php:77
<pre>PHP Fatal Error &#039;yii\base\ErrorException&#039; with message &#039;SOAP-ERROR: Parsing WSDL: Couldn&#039;t load from &#039;wrong_wsdl&#039; : failed to load external entity &quot;wrong_wsdl&quot;
&#039; 

in Client.php:77

Stack trace:
#0 vendor/codeception/codeception/codecept(42): Codeception\Application-&gt;run()
#1 {main}</pre>
```
> Test source code

```php
/**
 * @expectedException MyException
 */
public function testConnectFail()
{
    $client = new Client();
    $client->connect();
}
```
### Details

* Codeception version: 2.3.7
* PHP Version: 7.1.11
* Operating System: Ubuntu 17.10
* Installation type: Composer
* Suite configuration:

```yml
class_name: UnitTester
modules:
    enabled:
        - Asserts
        - \Helper\Unit
```

**Comment by Naktibalda** [comment]
It wasn't \SoapFault, try to catch \Error

**Comment by nnrudakov** [comment]
It doesn't matter. \SoapFault or \Exception wherever

**Comment by Naktibalda** [comment]
\Error is not \Exception http://php.net/manual/en/language.errors.php7.php#language.errors.php7.hierarchy

**Comment by nnrudakov** [comment]
In this case test failed
```bash
Failed asserting that exception of type "SoapFault" matches expected exception
```
Catch block not executed.

**Comment by Naktibalda** [comment]
Because it wasn't a SoapFault.

**Comment by nnrudakov** [comment]
When I run application (not tests) I've got exactly exception I want without any additional fatal messages.
And Codeception confirm it showing that test is OK.
I think cause in [error handler](https://github.com/Codeception/Codeception/blob/2.3/src/Codeception/Subscriber/ErrorHandler.php#L96). Application (Yii2) handler caught error and write it in log, but Codecetion handler output it.

**Comment by DavertMik** [comment]
Yeah looks like a conflict of error handlers. Codeception's error handler doesn't add any `<pre>` so I suspect Yii runs it error handler first...

``` 
in Client.php:77
<pre>PHP Fatal Error &#039;yii\base\ErrorException&#039; with message &#039;SOAP-ERROR: Parsing WSDL: Couldn&#039;t load from &#039;wrong_wsdl&#039; : failed to load external entity &quot;wrong_wsdl&quot;
&#039; 

in Client.php:77
```
Looks like the Yii's error handler doing some job that prevents us to handle that error. 
@samdark any ideas?

**Comment by artmnv** [comment]
This is because of php bug: https://bugs.php.net/bug.php?id=70469
It will be fixed in php 7.3. Now you can use ugly workaround: https://gist.github.com/artmnv/f15a47e63efb1f4bd995988fe2996919
